util/device_sw_utils/extract_sw_logs.py

```diff
@@ -121,6 +121,8 @@ def get_addr_strings(ro_contents):
         base_addr, size, data = ro_content
         while (str_start < size):
             str_end = data.find(b'\0', str_start)
```

Comment:
Even if we do find a nul byte, that doesn't necessarily mean the entity is a string.

I'd guess that the way this tool is used is to extract a set of addresses of logging strings so that simulations don't have to decode the bitstream out of the uart, but rather can just look for magic addresses written into a logging buffer and then convert those back to text as part of presenting test results.  Is that correct?

Reply:
Yes that's right.